src/main/java/org/kohsuke/github/GHRepositoryDiscussion.java

```diff
@@ -0,0 +1,210 @@
+package org.kohsuke.github;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.kohsuke.github.internal.EnumUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Date;
+import java.util.Objects;
+
+/**
+ * A discussion in the repository.
+ * <p>
+ * This is different from Teams discussions (see {@link GHDiscussion}).
+ * <p>
+ * The discussion event exposes (more or less) the GraphQL object directly. The new Discussions API is only available
+ * through GraphQL so for now you cannot execute any actions on this object.
+ *
+ * @author Guillaume Smet
+ * @see <a href="https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#discussion">The GraphQL
+ *      API for Discussions</a>
+ */
+public class GHRepositoryDiscussion {
```

Comment:
Why not this? It should remove a bunch of fields and code from this file.  

```suggestion
public class GHRepositoryDiscussion extends GHObject {
```

At very least you can extend `GitHubInteractiveObject`.

Reply:
Yeah so initially I didn't want that because this object was not really "interactive" given you can't interact with it. But yeah, you're right, it's probably easier to have everything consistent. I'll push something to fix it.